src/behavior-router.ts

```diff
@@ -97,14 +99,19 @@ export class BehaviorRouter {
 				}
 
 				const handler = this.match(req);
-				const cfEvent = convertToCloudFrontEvent(req, this.builder('viewer-request'));
 
 				if (!handler) {
 					res.statusCode = StatusCodes.NOT_FOUND;
 					res.end();
 					return;
 				}
 
+				const customHeaders = handler.distribution in this.cfResources ?
+					getCustomHeaders(this.cfResources[handler.distribution]):
+					{}
+
+				const cfEvent = convertToCloudFrontEvent(req, this.builder('viewer-request'), customHeaders);
```

Comment:
I'm not familiar with the behavior of custom headers yet? Do the custom headers need to apply from the initial event through the entire lifecycle? Or do they only apply once the request transits to the origin?

Reply:
They get applied to a different part of the request object. I need to recheck to be honest. The current implementation here however is definatley wrong. Misread the CF event. So will need to refactor this in either case. Given I do not think the headers ever get merged. But if they do it is certainly not here :thinking: